src/core/source_buffers/segment_inventory.ts

```diff
@@ -23,51 +23,72 @@ import {
   Period,
   Representation,
 } from "../../manifest";
-import { convertToRanges } from "../../utils/ranges";
 import takeFirstSet from "../../utils/take_first_set";
 
-const { MAX_MANIFEST_BUFFERED_DIFFERENCE,
+const { MAX_MANIFEST_BUFFERED_START_END_DIFFERENCE,
+        MAX_MANIFEST_BUFFERED_DURATION_DIFFERENCE,
         MINIMUM_SEGMENT_SIZE } = config;
 
 interface IBufferedChunkInfos { adaptation : Adaptation;
                                 period : Period;
                                 representation : Representation;
                                 segment : ISegment; }
 
+// Stored information for each chunk
 export interface IBufferedChunk {
-  bufferedEnd : number|undefined; // Last inferred end in the SourceBuffer
+  bufferedEnd : number|undefined; // Last inferred end in the SourceBuffer this
+                                  // chunk ends at, in seconds
   bufferedStart : number|undefined; // Last inferred start in the SourceBuffer
-  end : number; // Supposed end the segment should reach
+                                    // this chunk starts at, in seconds
+  end : number; // Supposed end, in seconds, the chunk is expected to reach
   precizeEnd : boolean; // if `true` the `end` property is an estimate we have
-                        // enough confidence in.
+                        // enough confidence in (and thus `bufferedEnd` should
+                        // be very close to that value when that chunk has been
+                        // pushed).
                         // if `false`, it is just a guess based on segment
                         // information. in that case, we might update that
                         // value the next time we check the state of the
                         // buffered segments.
   precizeStart : boolean; // if `true` the `start` is an estimate we have enough
-                          // confidence in.
+                          // confidence in (and thus `bufferedStart` should
+                          // be very close to that value when that chunk has
+                          // been pushed).
                           // if `false`, it is just a guess based on segment
                           // information. in that case, we might update that
                           // value the next time we check the state of the
                           // buffered segments.
   infos : IBufferedChunkInfos; // Information on what this segment is in terms
                                // of content.
-  isCompleteSegment : boolean; // If true, the whole segment has been completely
-                               // pushed. If false, it is either still pending
-  start : number; // Supposed start the segment should start from
+  isCompleteSegment : boolean; // If `true`, the whole segment has been
```

Comment:
Finally, maybe this attribute should be called differently. It does not signify that the segment is still loading and / or being pushed. As a not-rxplayer-friendly reader, you could think that the segment is not complete for another reason. However, the `isCompleteSegment` property seems to be used in the sense `"notFullyPushedSegment"`.